### ISSUE TITLE
gradient calculation in mini batch

### ISSUE THREAD

**Issue by mks0601** [rykov8/ssd_keras#8]
Hi again
I have a question about calculating gradient
In ssd_training.py, it seems you give loss for each batch, not a averaged loss of total batch.
Is this correct?

**Comment by rykov8** [comment]
@mks0601 hi! Yes, this is correct, because `Keras` expects loss function to be a symbolic function that returns a scalar for each data-point. The actual optimized objective is the mean of the output array across all datapoints. It is stated [here is `Keras` docs](https://keras.io/objectives/).
